back/scripts/utils/dataframe_operation.py

```diff
@@ -13,13 +14,17 @@
 
 
 # Function to merge duplicate columns in a DataFrame
-def merge_duplicate_columns(df):
+def merge_duplicate_columns(df: pd.DataFrame, separator: str = " / ") -> pd.DataFrame:
     df.columns = df.columns.astype(str)
     duplicate_columns = df.columns[df.columns.duplicated(keep=False)]
+
     for col in duplicate_columns.unique():
-        cols_to_merge = df.filter(like=col, axis=1)
-        df[col] = cols_to_merge.apply(lambda x: " / ".join(x.dropna().astype(str)), axis=1)
-        df.drop(cols_to_merge.columns[1:], axis=1, inplace=True)
+        first_col_index = df.columns.to_list().index(col)
+        merged_serie = df[col].apply(lambda x: separator.join(x.dropna().astype(str)), axis=1)
```

Comment:
Peut-être des techniques de ce lien pourraient éviter le apply qu'il est toujours mieux d'éviter si on peut.
https://stackoverflow.com/questions/39291499/how-to-concatenate-multiple-column-values-into-a-single-column-in-pandas-datafra

Reply:
Rien de concluant, il existe plein de façon de concaténer des séries mais le cas particulier ici est le dropna que l'on veut appliquer de façon à ne pas supprimer des lignes entières.
Cela dit, le problème précédent est aussi dû au dropna…